File: GeneratedEntityReaderClasses/Components/PositionSeedComponent.cs
namespace GeneratedNoitaClasses;
public class PositionSeedComponent : NoitaComponentBase,  iNoitaType<PositionSeedComponent>{
	public NoitaBool Deleted { get; set; } = new();
	public NoitaBool Enabled { get; set; } = new();
	public NoitaString Tags { get; set; } = new();
	public NoitaFloat pos_x { get; set; } = new();
	public NoitaFloat pos_y { get; set; } = new();

    public void Read(NoitaStream s){
		Deleted.Read(s);
		Enabled.Read(s);
		Tags.Read(s);
		pos_x.Read(s);
		pos_y.Read(s);

    }
    public void Write(NoitaStream s){
		Deleted.Write(s);
		Enabled.Write(s);
		Tags.Write(s);
		pos_x.Write(s);
		pos_y.Write(s);

    }
}

File: GeneratedEntityReaderClassesPython/GeneratedClasses/Components/PositionSeedComponent.py
from dataclasses import dataclass, field
from BasicNoitaTypes import NoitaBool
from BasicNoitaTypes import NoitaString
from BasicNoitaTypes import NoitaFloat

from NoitaStream import NoitaStream
@dataclass
class PositionSeedComponent:
    Deleted:NoitaBool = field(default_factory=NoitaBool)
    Enabled:NoitaBool = field(default_factory=NoitaBool)
    Tags:NoitaString = field(default_factory=NoitaString)
    pos_x:NoitaFloat = field(default_factory=NoitaFloat)
    pos_y:NoitaFloat = field(default_factory=NoitaFloat)
    
    def Read(self, s:NoitaStream):
        self.Deleted.Read(s)
        self.Enabled.Read(s)
        self.Tags.Read(s)
        self.pos_x.Read(s)
        self.pos_y.Read(s)
    
    def Write(self, s:NoitaStream):
        self.Deleted.Write(s)
        self.Enabled.Write(s)
        self.Tags.Write(s)
        self.pos_x.Write(s)
        self.pos_y.Write(s)

